homeassistant/components/knx/__init__.py

```diff
@@ -183,19 +178,40 @@
     }
 )
 
+SERVICE_KNX_EXPOSURE_REGISTER_SCHEMA = vol.Any(
+    ExposeSchema.SCHEMA.extend(
```

Comment:
It's a risk to use a config schema as base for a service schema. The constants are different and may change independently.

Reply:
The intention was that if it was changed in one place (the config schema) the change is automatically applied to the service - as the service calls the exact same functions as the config 
`create_knx_exposure(self.hass, self.xknx, call.data)` from service and
`create_knx_exposure(hass, hass.data[DOMAIN].xknx, expose_config)` from config

Reply:
Yes, but that only works safely if the constants are the same, which they aren't.

Reply:
I don't understand. Which constants differ?

Reply:
`SERVICE_KNX_ATTR_ADDRESS` and `CONF_KNX_EXPOSE_ADDRESS`. They have the same value currently.

Reply:
Ah, ok. Would it be ok to just use the same constant here imported from .const?

Or should I refactor to pass only Optional[str] to create_knx_exposure instead of call.data / expose_config

Reply:
Linking to the same constant could be ok. The details of knx are fuzzy to me so I can't say better about the other suggestion.